feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

```diff
@@ -0,0 +1,127 @@
+package com.iteneum.office.presentation.ui
+
+import android.util.Log
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Call
+import androidx.compose.material.icons.outlined.Mail
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextIndent
+import androidx.compose.ui.unit.sp
+import com.iteneum.IntentActionsImpl
+import com.iteneum.OfficeInfoItemModel
+import com.iteneum.PermissionsProviderImpl
+import com.iteneum.designsystem.components.LpOutlinedButton
+import com.iteneum.designsystem.theme.LeasePertTheme
+import com.iteneum.office.R
+
+/**
+ * This class will handle the network states responses, i modification with the structure.
+ * @param onClickItem click back when you do it on an item
+ * @param onSuccess our event that send us to the next view
+ * TODO implement @param viewModel the view model we are using inside our view compose has to be injected
+ * /*viewModel: OfficeViewModel = hiltViewModel()*/
+ * @author Andres Ivan Medina
+ */
+@Composable
+fun OfficeView() {
+    val viewModel = OfficeViewModel(
+        IntentActionsImpl(
+            LocalContext.current, PermissionsProviderImpl(
+                LocalContext.current
+            )
+        )
+    )
+    LaunchedEffect(true) {
+        viewModel.loadOfficeInformationData()
+    }
+    val officeDataList = remember {
+        viewModel.dataOfficeInfo
+    }
+    OfficeUI(
```

Comment:
Check with Daniel how you need to use high order function .

Reply:
Can you elaborate more about this suggestion is not clear, 

to consult with dan which specific  consideration of use, over high order functions, implemented in here?